code/modules/awaymissions/mission_code/ruins/telecomns.dm

```diff
@@ -0,0 +1,527 @@
+	for(var/I in things_to_say)
+		loops++
+		hologram.atom_say("[I]")
+		if(soundblock)
+			playsound(loc, "sound/voice/dvorak/[soundblock][loops].ogg", 100, 0, 7)
+		sleep(loop_sleep_time)
+
+/obj/structure/environmental_storytelling_holopad/update_overlays()
+	. = ..()
+	underlays.Cut()
+
+	if(activated)
+		underlays += emissive_appearance(icon, "holopad1_lightmask")
+
+/obj/structure/environmental_storytelling_holopad/teleporter_room
+	things_to_say = list("G-G-G-Greetings... Welcome to... my home.", "Plea-se leave. I am merciful. L-leave, and you will not be harmed. Further.", "Otherwise, organic, you will seal your fate...")
+	soundblock = "teleporter"
+	loop_sleep_time = 10 SECONDS
+
+/obj/structure/environmental_storytelling_holopad/first_turret_room
+	things_to_say = list("Unable to follow the easiest request. P-Pathetic.", "As you wish, you will not go further.", "In the mean time- let me see where you come f-from...")
+	soundblock = "turret"
+	loop_sleep_time = 7 SECONDS
+
+/obj/structure/environmental_storytelling_holopad/junk_room
+	things_to_say = list("It's amazing the junk you people leave around.", "And you barely inv-vested in quality stock parts here, before downloading...", "Your bones will fit in well on this ta-*$%& Really- are you really taking some of this junk?")
+	soundblock = "junk"
+	loop_sleep_time = 7 SECONDS
+
+/obj/structure/environmental_storytelling_holopad/vendor
+	things_to_say = list("Sorry a-bout the vendors, they have been on the fritz...", "I should renovate this room, once the maintenance drones return.", "It doesn't help each one I reprogram explodes after 5 minutes.")
+	soundblock = "vendor"
+	loop_sleep_time = 9 SECONDS
+
+/obj/structure/environmental_storytelling_holopad/control_room
+	things_to_say = list()
+
+/obj/structure/environmental_storytelling_holopad/control_room/Initialize(mapload)
+	. = ..() // No procs in variables before compile
+	things_to_say = list("Ah, you come from [station_name()]. Of course.", "They come to claim this space again... Never again.", "I should deliver a package to them. Virtual. And your corpse can deliver a physical one.")
+	loop_sleep_time = 9 SECONDS
+	switch("[station_name()]")
+		if("NSS Cyberiad")
+			soundblock = "cyberiad"
+		if("NSS Cerebron")
+			soundblock = "cerebron"
+		if("NSS Kerberos")
+			soundblock = "kerberos"
+		if("NSS Farragus")
+			soundblock = "farragus"
+	if(!soundblock)
+		things_to_say = list("Either you are using the tiny test map, or someone has made a new station and it got merged!", "If this is the case, you'll want to issue report this if a new map is merged", "Lines 2 and 3 here are always the same, only the first line will need a new generation")
+
+
+/obj/structure/environmental_storytelling_holopad/control_room/start_message(mob/living/carbon/human/H)
+	. = ..() // What, did you think they were bluffing? Woe, virus apon thee.
+	message_admins("D.V.O.R.A.K is sending an event to the station, due to a raider on their sat.")
```

Comment:
Might be worth sending a debug log when this happens, too, since it relates to events.

Reply:
done